### PR TITLE
use selected network name and return provider immediately if not default

### LINKED ISSUE #20: Hardhat l2ethers provider is always set to localhost
l2ethers provider is always set to localhost, even when passing a network flag to hardhat

If I write a simple deploy script, l2ethers doesn't take into account the network I've selected, and will always deploy against the local hardhat node. I can interact when it does this. However, I can't deploy to Kovan testnet using l2ethers alone, and can't interact with contracts on Kovan testnet at all with l2ethers.

If I use hardhat deploy, it does take into account the network I've selected and I see the tx on your explorer, but interacting with it causes l2ethers to blow up (likely related to the first issue and interactions).

### PR DESCRIPTION
**Pull request by matthewlilley**
Closes #20

### CODE REVIEW

**Comment by smartcontracts** [comment]
Oh! Huh. I hadn't thought about this for some reason. But it clearly makes sense. LGTM! Will include in the next release (within 1-2 days).

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
     "@types/sinon-chai": "^3.2.5",
     "chai": "^4.2.0",
     "ethereum-waffle": "^3.0.0",
+    "ethers": "^5.0.0",
```
Comment (smartcontracts) [inline package.json:56]: What is this change used for?

Comment (matthewlilley) [reply package.json:56]: Force of habit to mirror peerDeps in devDeps for local development, but feel free to remove, I'm not sure there's a good reason in this case.

Comment (matthewlilley) [reply package.json:56]: I just noticed ethers is both a dependency and a peer dependency. Is that intentional?

Comment (smartcontracts) [reply package.json:56]: It needs to be a dev dependency and a peer dep. Shouldn't be a primary dep. That might be a bug.

Comment (smartcontracts) [inline package.json:56]:
```suggestion
```

Comment (smartcontracts) [reply package.json]: Removed for this PR. I'll take a look at why we made it part of the primary deps.

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**